### DISCUSSION TITLE
Testing muP.

### DISCUSSION THREAD

**Discussion by cloneofsimo** [cloneofsimo/min-max-gpt#4] [Show and tell]
# Initial Tests on muP running from 110M ~ 4.4B models.

Our goal is to see if results from [**TP-V5**](https://arxiv.org/abs/2203.03466) and [**muScaling**](https://arxiv.org/abs/2304.06875) results are reproducible, and test to see if muP provides stable scaling-law predictions. 

```bash
export WORLD_SIZE=$(nvidia-smi -L | wc -l)
#!/bin/bash

lrs=(1e-4 2e-4 5e-4 1e-3)
widths=(16 32 64 128 256)

for lr in "${lrs[@]}"; do
    for width in "${widths[@]}"; do
        run_name="lr_${lr}_width_${width}"
        deepspeed --num_gpus $WORLD_SIZE run_trainer.py --lr $lr --width $width --run_name $run_name
    done
done
```

---

Current Status:

![image](https://github.com/cloneofsimo/min-max-gpt/assets/35953539/935c94a3-e682-4dcc-b3dd-e636a98405e8)

**Comment by cloneofsimo** [comment]
![image](https://github.com/cloneofsimo/min-max-gpt/assets/35953539/22504d8c-dd9a-4916-b4f1-9b13259f70a8)
---
Looks correct to me!

**Comment by xszheng2020** [comment]
Could you please share the loss curves? with respect to training steps

**Reply by cloneofsimo** [reply]
Yeah sure, here plots of some more recent runs
![image](https://github.com/cloneofsimo/min-max-gpt/assets/35953539/a0123a24-fc80-4f16-b00a-aa88d725fca3)
